WordPress/src/main/java/org/wordpress/android/ui/posts/editor/EditorTracker.kt

```diff
@@ -104,8 +109,17 @@ class EditorTracker @Inject constructor(
             TrackableEvent.UNDERLINE_BUTTON_TAPPED -> Stat.EDITOR_TAPPED_UNDERLINE
             TrackableEvent.REDO_TAPPED -> Stat.EDITOR_TAPPED_REDO
             TrackableEvent.UNDO_TAPPED -> Stat.EDITOR_TAPPED_UNDO
+            TrackableEvent.EDITOR_GUTENBERG_UNSUPPORTED_BLOCK_WEBVIEW_CLOSED ->
+                Stat.EDITOR_GUTENBERG_UNSUPPORTED_BLOCK_WEBVIEW_CLOSED
+            TrackableEvent.EDITOR_GUTENBERG_UNSUPPORTED_BLOCK_WEBVIEW_SHOWN ->
+                Stat.EDITOR_GUTENBERG_UNSUPPORTED_BLOCK_WEBVIEW_SHOWN
         }
 
-        analyticsTrackerWrapper.track(currentStat, mapOf("editor" to editorName))
+        if (properties.isEmpty()) {
+            analyticsTrackerWrapper.track(currentStat, mapOf("editor" to editorName))
+        } else {
+            properties.put("editor", editorName)
+            analyticsTrackerWrapper.track(currentStat, properties)
+        }
```

Comment:
Not a big deal, but instead of having this if-else block, could we just do the following regardless of whether the property map is empty?

```
properties["editor"] = editorName // or properties.put("editor", editorName)
analyticsTrackerWrapper.track(currentStat, properties)
```

We could take this even further and avoid mutating the map by accepting a non-mutable map as the function parameter (instead of the mutable map the function currently requires) and creating a new map with the proper "editor" value:

```
val updatedProperties = properties + ("editor" to editorName)
analyticsTrackerWrapper.track(currentState, updatedProperties)
```

Reply:
You are Kotlin ninja, text for the tips! 👍

Reply:
Fixed.